Remove references to the Observer

SiAdcock opened the pull request:
<!-- See https://github.com/guardian/recommendations/blob/main/pull-requests.md for recommendations on raising and reviewing pull requests. -->

## What does this change?

- Refresh wording on [Guardian Media Group](https://www.theguardian.com/about/organisation) page under _Our Structure_ for a more dynamic portrayal, and remove references to Observer. 
- Remove _History of the Observer_ card from [The Scott Trust](https://www.theguardian.com/about/history) page

## Screenshots


| Before      | After      |
| ----------- | ---------- |
| ![before-gmg][] | ![after-gmg][] |
| ![before-st][] | ![after-st][] |

[before-st]: https://github.com/user-attachments/assets/1d295cf8-8f84-4c3f-a212-7af88a0ae88d
[after-st]: https://github.com/user-attachments/assets/7e25447b-f596-4ebe-a73a-736aac36ebc4
[before-gmg]: https://github.com/user-attachments/assets/ab390fce-0c0d-4983-8e97-ac46549a04b2
[after-gmg]: https://github.com/user-attachments/assets/27586427-606f-47e1-a78d-de0fb0303da3

petternitter reviewed (commented):
The before/after screenshots for the updates to the _Our structure_ copy seem to have been swapped around; just confirming that's the case and that what's currently screenshotted as _before_ is actually the _after_? The code suggests that's the case.

SiAdcock commented:
Thanks @petternitter. I have: 

- removed the UK edition card from the about and journalism pages 
- removed "GNM" from the _Executive committee_ card
- rejigged the cards on the Journalism page, removed the Observer card, added the Feast card

I couldn't easily reinstate the body text to the _Executive committee_ card, it looked broken at mobile width. We would need to make a new card variant to support this, which we can discuss.

Unless there's any last minute / urgent fixes, I'm gonna suggest this is done and ready to go live on Tuesday.

<img width="340" alt="Screenshot 2025-04-17 at 13 04 37" src="https://github.com/user-attachments/assets/939aeef7-ae31-4891-be01-25b4fe4a57f4" />
